### ISSUE TITLE
Multiple leaders elected

### ISSUE THREAD

**Issue by adsharma** [adsharma/raft#1]
Setup: three participants - Alice, Bob and Charlie.

Alice Logs:

```
$ ./zre_raft/chat.py -n alice
using Python native module
bob cce7aeb7-9c55-431b-85c1-3a89bc8624f1 entered
bob cce7aeb7-9c55-431b-85c1-3a89bc8624f1 joined raft
alice raft: Lost Leader: None
bob raft: 
cce7aeb7-9c55-431b-85c1-3a89bc8624f1 exit 
alice raft: Lost Leader: None
bob 2d64670e-48e3-4e35-99c5-359738078b0a entered
bob 2d64670e-48e3-4e35-99c5-359738078b0a joined raft
charlie b0efbf70-9180-4758-9b95-d1bfe02d84ed entered
charlie b0efbf70-9180-4758-9b95-d1bfe02d84ed joined raft
alice raft: Lost Leader: None
4c0768b8461a4c33bdf582d286bf90dc: New Leader
charlie 5eaca74d-2ce4-4aa4-a5a9-790f17a00133 entered
charlie 5eaca74d-2ce4-4aa4-a5a9-790f17a00133 joined raft
b0efbf70-9180-4758-9b95-d1bfe02d84ed exit 
charlie raft: 
bob raft: 
alice raft: /status
4c0768b8-461a-4c33-bdf5-82d286bf90dc
consensus: ZREServer(_name='4c0768b8461a4c33bdf582d286bf90dc', _state=<simpleRaft.states.leader.Leader object at 0x7f8038adc820>, _log=[], _messageBoard=<simpleRaft.boards.memory_board.MemoryBoard object at 0x7f8038adccd0>, _neighbors=[UUID('2d64670e-48e3-4e35-99c5-359738078b0a'), UUID('5eaca74d-2ce4-4aa4-a5a9-790f17a00133')], _commitIndex=0, _currentTerm=5, _lastApplied=0, _lastLogIndex=0, _lastLogTerm=None)
```

Charlie Logs:

```
$ ./zre_raft/chat.py -n charlie
using Python native module
bob 2d64670e-48e3-4e35-99c5-359738078b0a entered
bob 2d64670e-48e3-4e35-99c5-359738078b0a joined raft
alice 4c0768b8-461a-4c33-bdf5-82d286bf90dc entered
alice 4c0768b8-461a-4c33-bdf5-82d286bf90dc joined raft
charlie raft: Accepted new leader: 4c0768b8461a4c33bdf582d286bf90dc
Lost Leader: 4c0768b8461a4c33bdf582d286bf90dc
5eaca74d2ce44aa4a5a9790f17a00133: New Leader
Lost Leader: 4c0768b8461a4c33bdf582d286bf90dc
5eaca74d2ce44aa4a5a9790f17a00133: New Leader
charlie raft:
charlie raft: /status
5eaca74d-2ce4-4aa4-a5a9-790f17a00133
consensus: ZREServer(_name='5eaca74d2ce44aa4a5a9790f17a00133', _state=<simpleRaft.states.leader.Leader object at 0x7f28bc501580>, _log=[], _messageBoard=<simpleRaft.boards.memory_board.MemoryBoard object at 0x7f28bc64dc70>, _neighbors=[UUID('2d64670e-48e3-4e35-99c5-359738078b0a'), UUID('4c0768b8-461a-4c33-bdf5-82d286bf90dc')], _commitIndex=0, _currentTerm=5, _lastApplied=0, _lastLogIndex=0, _lastLogTerm=None)
```

And then Bob alternates between the two leaders:

```
Accepted new leader: 4c0768b8461a4c33bdf582d286bf90dc
Accepted new leader: 5eaca74d2ce44aa4a5a9790f17a00133
Accepted new leader: 4c0768b8461a4c33bdf582d286bf90dc
Accepted new leader: 5eaca74d2ce44aa4a5a9790f17a00133
Accepted new leader: 4c0768b8461a4c33bdf582d286bf90dc
Accepted new leader: 5eaca74d2ce44aa4a5a9790f17a00133
Accepted new leader: 4c0768b8461a4c33bdf582d286bf90dc
```

**Comment by adsharma** [comment]
Suspected cause:

Once a new leader is accepted and we transition to `Follower` state, the timer is not cancelled.

**Comment by adsharma** [comment]
Fixed by  ce36016
